tests/tests.rs

```diff
@@ -801,3 +801,88 @@ fn test_issue_118() {
 
     let _ = Foo::<u32>::builder().build();
 }
+
+#[test]
+fn test_preinitialized_fields() {
+    #[derive(Debug, PartialEq, TypedBuilder)]
+    struct Foo {
+        x: i32,
+        #[builder(via_mutators)]
+        y: i32,
+        #[builder(via_mutators = 2)]
+        z: i32,
+        #[builder(via_mutators(init = 2))]
+        w: i32,
+    }
+
+    let foo = Foo::builder().x(1).build();
+    assert_eq!(foo, Foo { x: 1, y: 0, z: 2, w: 2 });
+}
+
+#[test]
+fn test_mutators_item() {
+    #[derive(Debug, PartialEq, TypedBuilder)]
+    #[builder(mutators(
+        fn inc_x(self: (x)) {
+            self.x += 1;
+        }
+        fn inc_x_by(self: (x), x: i32) {
+            self.x += x;
+        }
+        fn inc_preset(self) {
+            self.y += 1;
+            self.z += 1;
+            self.w += 1;
+        }
+        fn inc_y_by_x(self:(x)) {
```

Comment:
WTF? I thought the `: (x)` syntax means that only `self.x` is accessible - how come you can access `y` in this mutator?

Also - I'm not sure I like this syntax. It doesn't look like Rust. I think the proper way would be to use an attribute:

```rust
#[mutator(access(x, y))]
fn inc_y_by_x(self) {
```
Exact syntax and naming to be bikeshed.

Or maybe something like this:

```rust
fn inc_y_by_x(self: MutatorAccess!(x, y)) {
```

Where `MutatorAccess` will fake a struct that has x and y.

Reply:
> WTF? I thought the `: (x)` syntax means that only `self.x` is accessible - how come you can access `y` in this mutator?

As `via_mutators` are always available, I thought it wouldn't be necessary to specify them here. I.e. I had the (x) extend the Fields available.


> Also - I'm not sure I like this syntax. It doesn't look like Rust.

It kindof isn't, because it would mean self is a tuple struct, I can understand your hesitation here.

> I think the proper way would be to use an attribute:
> 
> ```rust
> #[mutator(access(x, y))]
> fn inc_y_by_x(self) {
> ```
> Exact syntax and naming to be bikeshed.
> 
> Or maybe something like this:
> 
> ```rust
> fn inc_y_by_x(self: MutatorAccess!(x, y)) {
> ```
> 
> Where `MutatorAccess` will fake a struct that has x and y.

I'm open to whatever you prefer here, wouldn't change much implementation wise.

Reply:
alright, will update. I think this will make the parsing probably cleaner even.

Reply:
> ```rust
> #[mutator(dependencies = [x])]
> fn inc_y_by_x(self) {
> ```

Naming wise, I also wondered about calling it `requires` or `required_fields`.

Reply:
Also, we could use `requires(x, y, z)` instead of `requires = [x, y, z]`. Not really a preference.

Reply:
I prefer `requires = [x, y, z]`. `requires(x, y, z)` makes it seem like `x`, `y` and `z` are sub-settings of `requires`, just like `requires` is a sub-setting of `mutator`.